packages/@tinacms/graphql/src/schema/index.ts

```diff
@@ -139,9 +139,7 @@ export class TinaSchema {
   }
   public getCollectionByFullPath = (filepath: string) => {
     const collection = this.getCollections().find((collection) => {
-      return filepath
-        .replace('\\', '/')
-        .startsWith(collection.path.replace(/\/?$/, '/'))
+      return filepath.replace('\\', '/').startsWith(collection.path)
```

Comment:
So the background on this change was that we're current grabbing the first collection that matches the start of a collection's path.

So if I have 2 collections, with the following paths:
`foo` & `foo2`

When I fetch a document at `foo/my-post.md`, it may grab the wrong collection (`foo2`).

This change checks the document against `foo/**/*` instead of `foo**/*`

Reply:
Looking into this now.

Reply:
@jamespohalloran Pushed up a fix! Thanks